### ISSUE TITLE
[contract] startExit feature

### ISSUE THREAD

**Issue by boolafish** [omisego/plasma-cash#15]

**Comment by peara** [comment]
Hi, I was reading your implementation of Plasma Cash and noticed this:
https://github.com/omisego/plasma-cash/blob/master/plasma_cash/root_chain/contracts/RootChain/RootChain.sol#L90
It should be:
`prevTxList[3].toUint() == txList[2].toUint()`
isn't it?

**Comment by mkchungs** [comment]
Hi @bun919tw,

A follow-up question about your `startExit` implementation: Don't you also need to check whether tx is signed by the prevTx owner? More specifically, check `prevTxList[3].toAddress ==  ecrecover(tx_Msg, txList[4].toData())`?

PS: We’ve implemented [our version](https://rinkeby.etherscan.io/address/0x881544e0b2e02a79ad10b01eca51660889d5452b#code) of `startExit`, following the specs we published at [ethresearch](https://ethresear.ch/t/plasma-cash-with-sparse-merkle-trees-bloom-filters-and-probabilistic-transfers "PlasmaCash with Bloom Filters and Probabilistic Transfers"). 
Any feedback/suggestions would be greatly appreciated!

**Comment by boolafish** [comment]
I actually believe that previous tax can be different owner, isn’t it?

**Comment by bun919tw** [comment]
Hi @mkchungs 
Yes, you are right. We didn't check for the sender right now and will do this in next PR.
Congratulations on your work!! It's outstanding in my opinion!!

**Comment by mkchungs** [comment]
@boolafish yes,  previous tx can have different owner than the current tx. But I think current tx should be signed by the previous tx owner, right? 

Also we augmented our `TX` to include `PrevOwner` so that each transaction can be "self-verifiable".

**Comment by boolafish** [comment]
@mkchungs ur right! My brain is not working well these days ha.
